contrib/kind.sh

```diff
@@ -1127,6 +1142,149 @@ add_dns_hostnames() {
+      podman save -o /tmp/image.tar "${1}"
+      kind load image-archive /tmp/image.tar --name "${KIND_CLUSTER_NAME}"
+    else
+      kind load docker-image "${1}" --name "${KIND_CLUSTER_NAME}"
+    fi
+  fi
+}
+
+install_dnsnameresolver_images() {
+  install_image ${COREDNS_WITH_OCP_DNSNAMERESOLVER}
+  install_image ${DNSNAMERESOLVER_OPERATOR}
+}
+
+install_dnsnameresolver_operator() {
+  pushd /tmp/coredns-ocp-dnsnameresolver/operator
+  
+  # Before installing DNSNameResolver operator, update the args so that the operator
+  # is configured with the correct values.
+  sed -i -e 's/^\(.*--coredns-namespace=\).*/\1kube-system/' \
+    -e 's/^\(.*--coredns-service-name=\).*/\1kube-dns/' \
+    -e 's/^\(.*--dns-name-resolver-namespace=\).*/\1ovn-kubernetes/' \
+    -e 's/^\(.*--coredns-port=\).*/\153/' config/default/manager_auth_proxy_patch.yaml
+
+  make install
+  make deploy IMG=${DNSNAMERESOLVER_OPERATOR}
+  popd
+}
+
+update_clusterrole_coredns() {
+  original_clusterrole=$(kubectl get clusterrole system:coredns -oyaml)
+  echo "Original CoreDNS clusterrole:"
+  echo "${original_clusterrole}"
+  additional_permissions=$(printf '%s' '
+- apiGroups:
+  - network.openshift.io
+  resources:
+  - dnsnameresolvers
+  verbs:
+  - get
+  - list
+  - watch
+- apiGroups:
+  - network.openshift.io
+  resources:
+  - dnsnameresolvers/status
+  verbs:
+  - update
+  - get
+  - patch
+')
+  updated_clusterrole="${original_clusterrole}${additional_permissions}"
+  echo "Patched CoreDNS clusterrole:"
+  echo "${updated_clusterrole}"
+  printf '%s' "${updated_clusterrole}" | kubectl apply -f -
+}
+
+add_ocp_dnsnameresolver_to_coredns_config() {
+  original_corefile=$(kubectl get -n=kube-system configmap/coredns -o=jsonpath="{.data['Corefile']}")
+  if ! grep -wq "ocp_dnsnameresolver" ${original_corefile}; then
+    echo "Original CoreDNS Corefile:"
+    echo "${original_corefile}"
+    updated_corefile=$(
+      printf '%s' "${original_corefile}" | sed -e 's/^\(.*\)\(forward.*\)/\1ocp_dnsnameresolver {\n\1   namespaces ovn-kubernetes\n\1}\n\1\2/'
+    )
+    echo "Patched CoreDNS Corefile:"
+    echo "${updated_corefile}"
+    printf '%s' "${updated_corefile}" > /tmp/Corefile.json
+    updated_coredns=$(kubectl create configmap coredns -n=kube-system --from-file=Corefile=/tmp/Corefile.json -oyaml --dry-run=client)
+    echo "Patched CoreDNS config:"
+    echo "${updated_coredns}"
+    printf '%s' "${updated_coredns}" | kubectl apply -f -
+  fi
+}
+
+update_coredns_deployment_image() {
+  original_coredns_deploy=$(kubectl get -n=kube-system deploy/coredns -oyaml)
+  echo "Original CoreDNS deployment:"
+  echo "${original_coredns_deploy}"
+  updated_coredns_deployment=$(
+    printf '%s' "${original_coredns_deploy}" | sed -e "s|^\(.*image: \).*|\1${COREDNS_WITH_OCP_DNSNAMERESOLVER}|"
+  )
+  echo "Patched CoreDNS deployment:"
+  echo "${updated_coredns_deployment}"
+  printf '%s' "${updated_coredns_deployment}" | kubectl apply -f -
```

Comment:
I think can be replaced with
`kubectl -n=kube-system set image deploy/coredns coredns=${COREDNS_WITH_OCP_DNSNAMERESOLVER}`